templates/new/web_dodo.py.txt

```diff
@@ -10,69 +10,166 @@ IMAGE = "%s/%s:%s" % (CONFIG["company"], CONFIG["image"], CONFIG["version"])
 def task_build():
     """Build the Docker image for the current project"""
     return {{
-        'actions': ['docker build --tag=%s .' % IMAGE]
+        'actions': ['docker build --tag=%s .' % IMAGE],
+        'params': [
+            {{
+                'name': 'assembler',
+                'short': 'a',
+                'long': 'assembler',
+                'type': str,
+                'default': "0"
+            }}
+        ]
     }}
 
 def task_remove():
     """Remove the Docker image for the current project"""
     return {{
-        'actions': ['docker rmi %s -f' % IMAGE]
+        'actions': ['docker rmi %s -f' % IMAGE],
+        'params': [
+            {{
+                'name': 'assembler',
+                'short': 'a',
+                'long': 'assembler',
+                'type': str,
+                'default': "0"
+            }}
+        ]
     }}
 
 def task_dev():
     """Run the main task for the project"""
     return {{
-        'actions': ["docker run -p 8080:8080 --volume \"%s/\":/app %s python3 -m %s" % (CONFIG["volume_path"], IMAGE, PACKAGE_PATH)]
+        'actions': ["docker run -p 8080:8080 --volume \"%s/\":/app %s python3 -m %s --assembler \"%s\"" % (CONFIG["volume_path"], IMAGE, PACKAGE_PATH, "%(assembler)s")],
+        'params': [
+            {{
+                'name': 'assembler',
+                'short': 'a',
+                'long': 'assembler',
+                'type': str,
+                'default': "0"
+            }}
+        ]
     }}
 
 def task_prod():
     """Run the main task inside a Docker container for use in production """
     return {{
-        'actions': ["docker run -p 8080:8080 %s python3 -m %s" % (IMAGE, PACKAGE_PATH)],
-        'task_dep': ["build"]
+        'actions': ["docker run -p 8080:8080 %s python3 -m %s --assembler \"%s\"" % (IMAGE, PACKAGE_PATH, "%(assembler)s")],
+        'task_dep': ["build"],
+        'params': [
+            {{
+                'name': 'assembler',
+                'short': 'a',
+                'long': 'assembler',
+                'type': str,
+                'default': "0"
+            }}
+        ]
     }}
 
 def task_train():
     """Run training mode inside the container"""
     output_path = CONFIG["volume_path"] + "/output"
-    data_path = CONFIG["volume_path"] + "/data"
+    data_path = CONFIG["volume_path"] + "/input"
 
     return {{
-        'actions': ["docker run --volume \"%s\":/app/output --volume \"%s\":/app/data %s python3 -m %s --mode train" % (output_path, data_path, IMAGE, PACKAGE_PATH)]
+        'actions': ["docker run --volume \"%s\":/app/output --volume \"%s\":/app/input %s python3 -m {project_name} --mode train --assembler \"%s\"" % (output_path, data_path, IMAGE, "%(assembler)s")],
+        'params': [
```

Comment:
Rather than copy paste the params code multiple times can we make it a variable and define it once?

Reply:
Done